Carnap/src/Carnap/Languages/PurePropositional/Logic/Rules.hs

```diff
@@ -44,7 +44,7 @@ premConstraint (Just prems) sub | theinstance `elem` prems = Nothing
     where theinstance = pureBNF . applySub sub $ (SA (phin 1) :|-: SS (phin 1))
           project = view rhs
 
-dischargeConstraint n ded lhs sub | and (map (`elem` forms) lhs') = Nothing
+dischargeConstraint n ded lhs sub | all (`elem` forms) lhs' && all (`elem` lhs') forms = Nothing
```

Comment:
Here's where we now require that the explicitly cited dependencies are *exactly* the ones that show up on the LHS of the derived sequent.